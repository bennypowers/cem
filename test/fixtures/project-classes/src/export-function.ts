export function simple() {}
/** description */
export function described() {}
/** @summary summary */
export function summarized() {}
export function async() {}
export function *generator() {}

export const arrowsimple = () => {}
/** arrow description */
export const arrowdescribed = () => {}
/** @summary arrow summary */
export const arrowsummarized = () => {}
export const arrowasync = () => {}

export function param(a: string): string {}
/**
 * description
 * @param a param description
 * @return return description
 */
export function describedparam(a: string): string {}
export function rest(...as: string[]): string {}
/**
 * description
 * @param as rest description
 * @return return description
 */
export function describedrest(...as: string[]): string {}
