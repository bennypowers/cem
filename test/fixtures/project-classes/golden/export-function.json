{
  "schemaVersion": "1.0.0",
  "modules": [
    {
      "kind": "javascript-module",
      "path": "src/export-function.js",
      "declarations": [
        {
          "name": "simple",
          "kind": "function"
        },
        {
          "name": "described",
          "description": "description",
          "kind": "function"
        },
        {
          "name": "summarized",
          "summary": "summary",
          "kind": "function"
        },
        {
          "name": "async",
          "kind": "function"
        },
        {
          "name": "generator",
          "kind": "function"
        },
        {
          "name": "arrowsimple",
          "kind": "function"
        },
        {
          "name": "arrowdescribed",
          "description": "arrow description",
          "kind": "function"
        },
        {
          "name": "arrowsummarized",
          "summary": "arrow summary",
          "kind": "function"
        },
        {
          "name": "arrowasync",
          "kind": "function"
        },
        {
          "parameters": [
            {
              "name": "a",
              "type": {
                "text": "string"
              }
            }
          ],
          "name": "param",
          "kind": "function"
        },
        {
          "parameters": [
            {
              "name": "a",
              "description": "param description",
              "type": {
                "text": "string"
              }
            }
          ],
          "return": {
            "description": "return description"
          },
          "name": "describedparam",
          "description": "description",
          "kind": "function"
        },
        {
          "parameters": [
            {
              "name": "as",
              "type": {
                "text": "string[]"
              },
              "rest": true
            }
          ],
          "name": "rest",
          "kind": "function"
        },
        {
          "parameters": [
            {
              "name": "as",
              "description": "rest description",
              "type": {
                "text": "string[]"
              },
              "rest": true
            }
          ],
          "return": {
            "description": "return description"
          },
          "name": "describedrest",
          "description": "description",
          "kind": "function"
        }
      ],
      "exports": [
        {
          "kind": "js",
          "name": "simple",
          "declaration": {
            "name": "simple",
            "module": "/var/home/bennyp/Developer/cem/test/fixtures/project-classes/src/export-function.js"
          }
        },
        {
          "kind": "js",
          "name": "described",
          "declaration": {
            "name": "described",
            "module": "/var/home/bennyp/Developer/cem/test/fixtures/project-classes/src/export-function.js"
          }
        },
        {
          "kind": "js",
          "name": "summarized",
          "declaration": {
            "name": "summarized",
            "module": "/var/home/bennyp/Developer/cem/test/fixtures/project-classes/src/export-function.js"
          }
        },
        {
          "kind": "js",
          "name": "async",
          "declaration": {
            "name": "async",
            "module": "/var/home/bennyp/Developer/cem/test/fixtures/project-classes/src/export-function.js"
          }
        },
        {
          "kind": "js",
          "name": "generator",
          "declaration": {
            "name": "generator",
            "module": "/var/home/bennyp/Developer/cem/test/fixtures/project-classes/src/export-function.js"
          }
        },
        {
          "kind": "js",
          "name": "arrowsimple",
          "declaration": {
            "name": "arrowsimple",
            "module": "/var/home/bennyp/Developer/cem/test/fixtures/project-classes/src/export-function.js"
          }
        },
        {
          "kind": "js",
          "name": "arrowdescribed",
          "declaration": {
            "name": "arrowdescribed",
            "module": "/var/home/bennyp/Developer/cem/test/fixtures/project-classes/src/export-function.js"
          }
        },
        {
          "kind": "js",
          "name": "arrowsummarized",
          "declaration": {
            "name": "arrowsummarized",
            "module": "/var/home/bennyp/Developer/cem/test/fixtures/project-classes/src/export-function.js"
          }
        },
        {
          "kind": "js",
          "name": "arrowasync",
          "declaration": {
            "name": "arrowasync",
            "module": "/var/home/bennyp/Developer/cem/test/fixtures/project-classes/src/export-function.js"
          }
        },
        {
          "kind": "js",
          "name": "param",
          "declaration": {
            "name": "param",
            "module": "/var/home/bennyp/Developer/cem/test/fixtures/project-classes/src/export-function.js"
          }
        },
        {
          "kind": "js",
          "name": "describedparam",
          "declaration": {
            "name": "describedparam",
            "module": "/var/home/bennyp/Developer/cem/test/fixtures/project-classes/src/export-function.js"
          }
        },
        {
          "kind": "js",
          "name": "rest",
          "declaration": {
            "name": "rest",
            "module": "/var/home/bennyp/Developer/cem/test/fixtures/project-classes/src/export-function.js"
          }
        },
        {
          "kind": "js",
          "name": "describedrest",
          "declaration": {
            "name": "describedrest",
            "module": "/var/home/bennyp/Developer/cem/test/fixtures/project-classes/src/export-function.js"
          }
        }
      ]
    }
  ]
}